File: solutions/014.js
// Longest common prefix

//O(s) time where s = sum of all word lengths; O(1) space 

var longestCommonPrefix = function(strs) {
    if (strs.length === 0) {return ""}
    if (strs[0] === "") {return ""}
    let longest = ""
    for (let i = 0; i < strs[0].length; i++) {
        let match = strs[0][i]
        for (let j = 0; j < strs.length; j++) {
            if (strs[j][i] !== match) { 
                return longest
            }
        }
        longest += strs[0][i]
    }
    return longest
};

File: solutions/014.py
# Longest common prefix

class Solution:
    def longestCommonPrefix(self, words):
        i = 0
        prefix = ""
        if len(words) == 0:
            return ""
        while i < len(words[0]):
            letter = words[0][i]
            for x in words:
                if i >= len(x) or x[i] != letter:
                    return prefix
            prefix += words[0][i]
            i += 1
        return prefix
